src/Searching.js

```diff
@@ -17,29 +17,69 @@ limitations under the License.
 import EventIndexPeg from "./indexing/EventIndexPeg";
 import {MatrixClientPeg} from "./MatrixClientPeg";
 
-function serverSideSearch(term, roomId = undefined) {
-    let filter;
-    if (roomId !== undefined) {
-        // XXX: it's unintuitive that the filter for searching doesn't have
-        // the same shape as the v2 filter API :(
-        filter = {
-            rooms: [roomId],
+const SEARCH_LIMIT = 10;
+
+async function serverSideSearch(term, roomId = undefined, processResult = true) {
+    const client = MatrixClientPeg.get();
+
+    const filter = {
+        limit: SEARCH_LIMIT,
+    };
+
+    if (roomId !== undefined) filter.rooms = [roomId];
+
+    const body = {
+        search_categories: {
+            room_events: {
+                search_term: term,
+                filter: filter,
+                order_by: "recent",
+                event_context: {
+                    before_limit: 1,
+                    after_limit: 1,
+                    include_profile: true,
+                },
+            },
+        },
+    };
+
+    const response = await client.search({body: body});
+
+    if (processResult) {
+        const searchResult = {
+            _query: body,
+            results: [],
+            highlights: [],
         };
+
+        return client._processRoomEventsSearch(searchResult, response);
```

Comment:
They were separate methods first, but yeah I felt that there's too much repeated code. We can split them out again if you prefer it that way. Or perhaps we could lower the repeated code if one method calls the other one and we add a `process` suffix to it.

Reply:
Hmm, I think I prefer the second approach you mentioned, which sounds like a wrapper method to call `process` instead of a flag, which seems like it solves the changing return type issue.

Reply:
Done so.